File: mobile/ios/BiometricAuthModule.swift
// mobile/ios/BiometricAuthModule.swift
// iOS生物识别认证模块

import Foundation
import LocalAuthentication
import React

@objc(BiometricAuthModule)
class BiometricAuthModule: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func isBiometricAvailable(_ resolve: @escaping RCTPromiseResolveBlock,
                           rejecter reject: @escaping RCTPromiseRejectBlock) {
    let context = LAContext()
    var error: NSError?

    let available = context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error)

    if available {
      resolve(true)
    } else {
      resolve(false)
    }
  }

  @objc
  func authenticate(_ reason: String,
                   resolver resolve: @escaping RCTPromiseResolveBlock,
                   rejecter reject: @escaping RCTPromiseRejectBlock) {
    let context = LAContext()
    var error: NSError?

    guard context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: &error) else {
      reject("BIOMETRIC_NOT_AVAILABLE", "Biometric authentication not available", error)
      return
    }

    context.evaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, localizedReason: reason) { success, error in
      DispatchQueue.main.async {
        if success {
          resolve(["success": true])
        } else {
          reject("BIOMETRIC_AUTH_FAILED", "Biometric authentication failed", error)
        }
      }
    }
  }

  @objc
  func enrollBiometric(_ resolve: @escaping RCTPromiseResolveBlock,
                      rejecter reject: @escaping RCTPromiseRejectBlock) {
    // iOS不支持直接调用生物识别注册
    // 用户需要在系统设置中注册
    resolve(["success": true, "message": "Please enroll biometric in system settings"])
  }

  @objc
  func removeBiometric(_ resolve: @escaping RCTPromiseResolveBlock,
                      rejecter reject: @escaping RCTPromiseRejectBlock) {
    // iOS不支持直接移除生物识别
    resolve(["success": true, "message": "Please remove biometric in system settings"])
  }
}

File: mobile/android/app/src/main/java/com/xagent/BiometricAuthModule.kt
// mobile/android/app/src/main/java/com/xagent/BiometricAuthModule.kt
// Android生物识别认证模块

package com.xagent

import android.content.Context
import androidx.biometric.BiometricManager
import androidx.biometric.BiometricPrompt
import androidx.fragment.app.FragmentActivity
import com.facebook.react.bridge.*
import java.util.concurrent.Executor

class BiometricAuthModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String = "BiometricAuthModule"

    @ReactMethod
    fun isBiometricAvailable(promise: Promise) {
        try {
            val biometricManager = BiometricManager.from(reactApplicationContext)
            val canAuthenticate = biometricManager.canAuthenticate(
                BiometricManager.Authenticators.BIOMETRIC_STRONG or
                BiometricManager.Authenticators.BIOMETRIC_WEAK
            )

            val available = canAuthenticate == BiometricManager.BIOMETRIC_SUCCESS
            promise.resolve(available)
        } catch (e: Exception) {
            promise.reject("ERROR", e.message)
        }
    }

    @ReactMethod
    fun authenticate(reason: String, promise: Promise) {
        try {
            val activity = currentActivity as? FragmentActivity
                ?: throw Exception("Activity not available")

            val executor = Executor { command -> command.run() }
            val biometricPrompt = BiometricPrompt(
                activity,
                executor,
                object : BiometricPrompt.AuthenticationCallback() {
                    override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
                        super.onAuthenticationSucceeded(result)
                        val response = Arguments.createMap()
                        response.putBoolean("success", true)
                        promise.resolve(response)
                    }

                    override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
                        super.onAuthenticationError(errorCode, errString)
                        promise.reject("AUTH_ERROR", errString.toString())
                    }

                    override fun onAuthenticationFailed() {
                        super.onAuthenticationFailed()
                        promise.reject("AUTH_FAILED", "Authentication failed")
                    }
                }
            )

            val promptInfo = BiometricPrompt.PromptInfo.Builder()
                .setTitle("Biometric Authentication")
                .setSubtitle(reason)
                .setNegativeButtonText("Cancel")
                .build()

            biometricPrompt.authenticate(promptInfo)
        } catch (e: Exception) {
            promise.reject("ERROR", e.message)
        }
    }

    @ReactMethod
    fun enrollBiometric(promise: Promise) {
        promise.resolve(Arguments.createMap().apply {
            putBoolean("success", true)
            putString("message", "Please enroll biometric in system settings")
        })
    }

    @ReactMethod
    fun removeBiometric(promise: Promise) {
        promise.resolve(Arguments.createMap().apply {
            putBoolean("success", true)
            putString("message", "Please remove biometric in system settings")
        })
    }
}
